### PR TITLE
fix(connector): fix connector getConfig and validateConfig type

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Fix connector getConfig and validateConfig type.

## Testing

UT and tested locally.

### CODE REVIEW

File: packages/connector-alipay-native/src/index.test.ts [hunk 1]
```diff
@@ -15,18 +19,30 @@ describe('validateConfig', () => {
...
+  /**
+   * Assertion functions always need explicit annotations.
+   * See https://github.com/microsoft/TypeScript/issues/36931#issuecomment-589753014
+   */
+
   it('should pass on valid config', async () => {
-    await expect(
-      alipayNativeMethods.validateConfig(mockedAlipayNativeConfig)
-    ).resolves.not.toThrow();
+    const validator: ValidateConfig = alipayNativeMethods.validateConfig;
```
Comment (gao-sun) [inline packages/connector-alipay-native/src/index.test.ts:28]: the type of `validator` looks not right, maybe:
```suggestion
    const validator: typeof alipayNativeMethods.validateConfig = alipayNativeMethods.validateConfig;
```

File: packages/connector-aliyun-dm/src/index.ts [hunk 2]
```diff
@@ -19,24 +18,24 @@ import {
...
     const result = aliyunDmConfigGuard.safeParse(config);
 
     if (!result.success) {
       throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error);
     }
-  };
+  }
 
   // eslint-disable-next-line complexity
   public sendMessage: EmailSendMessageFunction = async (address, type, data, config) => {
```
Comment (gao-sun) [inline packages/connector-aliyun-dm/src/index.ts:34]: will we refactor this in the next PR?

**Review by gao-sun — APPROVED** [review]
looks amazing, thanks

File: packages/connector-alipay-native/src/index.test.ts (hunk 1 above)
Comment (gao-sun) [reply packages/connector-alipay-native/src/index.test.ts:28]: not blocking, add an issue and do it later

File: packages/connector-aliyun-dm/src/index.ts (hunk 2 above)
Comment (gao-sun) [reply packages/connector-aliyun-dm/src/index.ts:34]: cool

File: packages/connector-alipay-native/src/index.test.ts (hunk 1 above)
Comment (darcyYe) [reply packages/connector-alipay-native/src/index.test.ts:28]: [LOG-3661](https://linear.app/silverhand/issue/LOG-3661/connector-uts-validator-type-correction) to track this.
Resolved this to unblock upcoming changes.

File: packages/connector-smtp/src/index.ts [hunk 3]
```diff
@@ -14,22 +13,24 @@ import SMTPTransport from 'nodemailer/lib/smtp-transport';
...
     const result = smtpConfigGuard.safeParse(config);
 
     if (!result.success) {
       throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error);
     }
-  };
+  }
 
   public sendMessage: EmailSendMessageFunction = async (address, type, data) => {
     const config = await this.getConfig(this.metadata.id);
```
Comment (demonzoo) [inline packages/connector-smtp/src/index.ts:30]: Don't we need to test email config on the fly?
```typescript
public sendMessage: EmailSendMessageFunction = async (address, type, data) => {
  const config = config ?? (await this.getConfig(this.metadata.id));
...
```

**Pull request merged**